qupath-core-processing/src/main/java/qupath/opencv/tools/OpenCVTools.java

```diff
@@ -2651,6 +2651,69 @@ public static List<IndexedPixel> getMaskedPixels(Mat mat, Mat mask) {
 		}
 		return pixels;
 	}
+
+	/**
+	 * <p>Extract pixels from an image using a mask, limited to the specified channel index.</p>
+	 * <p>The mask can have one or multiple channels:</p>
+	 * <ul>
+	 *     <li>If the mask has only one channel, this channel is used.</li>
+	 *     <li>If the mask is multichannel, only the specified channel is used.</li>
+	 * </ul>
+	 * <p>The input and the mask must have the same dimensions.</p>
+	 *
+	 * @param input  the image from where the pixels should be extracted
+	 * @param mask  the mask to apply to the input
+	 * @param channel  the channel index (zero-based) to consider
+	 * @return a float array containing the values of the input filtered by the mask
+	 * @throws IllegalArgumentException when the input and the mask don't have the same dimensions
+	 */
+	public static float[] extractMaskedPixels(Mat input, Mat mask, int channel) {
```

Comment:
My fault, sorry... I see that it should be `extractMaskedFloats` for similarity to existing methods (since `extractPixels` takes an output array).